### PR TITLE
Clarify lack of guarantee in FIRWriteBatch completion

### PR DESCRIPTION
**Pull request by morganchen12**
Docs change from #478

### CODE REVIEW

File: Firestore/Source/Public/FIRWriteBatch.h [hunk 1]
```diff
@@ -98,7 +98,8 @@ FIR_SWIFT_NAME(WriteBatch)
  * Commits all of the writes in this write batch as a single atomic unit.
  *
  * @param completion A block to be called once all of the writes in the batch have been
- * successfully written to the backend as an atomic unit.
+ * successfully written to the backend as an atomic unit. This block is not guaranteed
+ * to execute if the device remains offline.
```
Comment (wilhuff) [inline Firestore/Source/Public/FIRWriteBatch.h]: This doesn't directly address the concern expressed in the underlying issue and doesn't directly caution users against using it.

Better would be something like:

This block _only_ runs while the device is online and the commit has completed against the server. The completion handler will not be called while the device is offline.

**Comment by mikelehen** [comment]
I think we should have this exact same verbiage on all of our other write methods (in FIRDocumentReference.h), no?

**Comment by wilhuff** [comment]
@morganchen12 Are you up for propagating this new improved verbiage to the other write methods (in transactions and document references)?

File: Firestore/Source/Public/FIRFirestore.h [hunk 2]
```diff
@@ -115,10 +115,11 @@ NS_SWIFT_NAME(Firestore)
  * and collections. Unlike other firestore access, data accessed with the transaction will not
  * reflect local changes that have not been committed. For this reason, it is required that all
  * reads are performed before any writes. Transactions must be performed while online. Otherwise,
- * reads will fail, and the final commit will fail.
+ * reads will fail, the final commit will fail, and the completion block will return an error.
  *
  * @param updateBlock The block to execute within the transaction context.
- * @param completion The block to call with the result or error of the transaction.
+ * @param completion The block to call with the result or error of the transaction. This
+ *     block will always run eventually.
```
Comment (morganchen12) [inline Firestore/Source/Public/FIRFirestore.h]: @wilhuff unsure if this is something we're formally guaranteeing.

Comment (wilhuff) [reply Firestore/Source/Public/FIRFirestore.h]: Within the scope of the process yes. If the OS restarts us then of course we can't promise that.

You could phrase slightly differently to put less emphasis on the eventuality and instead highlight that unlike other completion handlers for writes this one will run online or off.

**Comment by morganchen12** [comment]
Changes should be properly propagated now.

File: Firestore/Source/Public/FIRWriteBatch.h [hunk 3]
```diff
@@ -96,7 +96,10 @@ NS_SWIFT_NAME(WriteBatch)
  * Commits all of the writes in this write batch as a single atomic unit.
  *
  * @param completion A block to be called once all of the writes in the batch have been
- * successfully written to the backend as an atomic unit.
+ * successfully written to the backend as an atomic unit. This block will only execute
```
Comment (wilhuff) [inline Firestore/Source/Public/FIRWriteBatch.h]: minor nit: please use a hanging indent for `@param` blocks.

**Pull request merged**